pkg/kubelet/kubelet.go

```diff
@@ -762,8 +766,11 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 	return node, nil
 }
 
-// registerWithApiserver registers the node with the cluster master.
+// registerWithApiserver registers the node with the cluster master. It is safe to call multiple times.
 func (kl *Kubelet) registerWithApiserver() {
+	if kl.registrationCompleted {
```

Comment:
Do we need a lock around this bool? If folks read the func doc they might call it simultaneously from multiple go routines.